app/controllers/profiles_controller.rb

```diff
@@ -28,6 +28,21 @@ def update
     end
   end
 
+  def delete
+    @only_owner_gems = current_user.only_owner_gems
+    @multi_owner_gems = current_user.rubygems_downloaded - @only_owner_gems
+  end
+
+  def destroy
+    if User.authenticate(current_user.email, params[:user][:password]) && current_user.destroy
+      flash[:notice] = t '.successful_flash'
+      redirect_to_root
```

Comment:
This can be shortened to one line if you'll avoid `redirect_to_root`, which is just shortcut for `redirect_to root_path`.
```ruby
redirect_to root_path, notice: t '.successful_flash'
```